### ISSUE TITLE
dalmacija radio station does not work and looks, that only in radio.de addon.

### ISSUE THREAD

**Issue by expresik** [fivebanger/kodi_addons#15]
I use only few radio stations, but I see one of them stopped to work:
using ids - dalmacija is not working and antywawa, meloradio, eskarock works well

There is still on the list, it can be run from their website (via http://shoutcast.pondi.hr:8000/listen.pls), but don't start from addon.

Additionally I've checked, that it works well during radio.de website, but in site source I see https://shoutcast.pondi.hr:9000/stream, so - different port and little different link.

This radiostation exists twice on radio.de and in the addon and in both apperances there is the same situation.

Additionaly: in addon's recent.jsn file I've found: http://shoutcast.pondi.hr:8000/stream/1/ - and it works well, pasted into webbrowsers bar. But still cannot run the dalmacija radio station via addons GUI or json-rpc.

**Comment by fivebanger** [comment]
There is a problem with the website's certificate. This is the stream that gets reported by the underlying radio.net API for station ID `dalmacija`:
```
https://shoutcast.pondi.hr:9000/stream
```
When trying to resolve this URL by using Python's urlopen() inside the radio.de add-on, this error is returned (see Kodi log file):
```
2024-11-28 21:13:20.061 T:296534   error <general>: resolve_url error, url: https://shoutcast.pondi.hr:9000/;stream/1, error: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1145)>
```

I have tried to pass the URL anyway to Kodi, just ignoring this error. The result is that Kodi's `CCurlFile` also stops because of this certificate error when trying to load this stream:
```
2024-11-28 21:13:20.209 T:296530   error <general>: CCurlFile::CReadState::FillBuffer - (0x7f2e281129b0) Failed: SSL peer certificate or SSH remote key was not OK(60)
2024-11-28 21:13:20.209 T:296530   error <general>: CCurlFile::Open - <https://shoutcast.pondi.hr:9000/stream> Failed with code 0:
                                                   
2024-11-28 21:13:20.210 T:296530   debug <general>: CFileCache::Open - <https://shoutcast.pondi.hr:9000/stream> opening
2024-11-28 21:13:20.210 T:296530   debug <general>: CurlFile::Open - <https://shoutcast.pondi.hr:9000/stream>
2024-11-28 21:13:20.230 T:289289   debug <general>: ------ Window Init (DialogBusy.xml) ------
2024-11-28 21:13:20.260 T:296530   error <general>: CCurlFile::CReadState::FillBuffer - (0x7f2e2806b5e0) Failed: SSL peer certificate or SSH remote key was not OK(60)
2024-11-28 21:13:20.260 T:296530   error <general>: CCurlFile::Open - <https://shoutcast.pondi.hr:9000/stream> Failed with code 0:
                                                   
2024-11-28 21:13:20.260 T:296530   error <general>: CFileCache::Open - <https://shoutcast.pondi.hr:9000/stream> failed to open
2024-11-28 21:13:20.261 T:296530   error <general>: Init: Error opening file https://shoutcast.pondi.hr:9000/stream
2024-11-28 21:13:20.261 T:296530   error <general>: CAudioDecoder: Unable to Init Codec while loading file https://shoutcast.pondi.hr:9000/stream
```

So even when the add-on would ignore the certificate error, Kodi won't play this station. 

The very same applies for the second station `Radio Dalmacija` using ID `radiodalmacija` and returning stream address
```
https://shoutcast.pondi.hr:9000/;stream/1
```

Also this station is not playable using Kodi because of the same certificate error. I have no idea why those streams can get played in the browser directly but not using Kodi. But since Kodi is also refusing to play the given addresses
```
https://shoutcast.pondi.hr:9000/stream
https://shoutcast.pondi.hr:9000/;stream/1
```
there is nothing I can do, not even providing any workaround.

**Comment by fivebanger** [comment]
Additional information (just found out):

When providing the address
```
http://shoutcast.pondi.hr:9000/stream
```
instead of
```
https://shoutcast.pondi.hr:9000/stream
```
(pls. note https vs. http) Kodi would play the station. Neither the radio.de add-on, nor CCurlFile reports any issue. So best guess: radio.net is providing a wrong stream URL, using https instead of http which leads to certificate error in that case. The same applies for stream address
```
https://shoutcast.pondi.hr:9000/;stream/1
```
As a workaround you can add the stream URLs as custom stations like described [here](https://github.com/fivebanger/kodi_addons/tree/master/plugin.audio.radiode#custom-stations).

**Comment by expresik** [comment]
Thanks for your reply, I'll try the workaround.
However I sent an email to broadcaster with the info.

**Comment by expresik** [comment]
As for now, main dalmacija listener, my wife, told me, that she can wait. Unfortunately there is no response from broadcaster yet. Where do you plan to release next version?

**Comment by fivebanger** [comment]
Fixed with v1.1.6

**Comment by expresik** [comment]
tested on my side - works like a charm, thank you.
